### DISCUSSION TITLE
WG_MTU does not adjust server mtu size

### DISCUSSION THREAD

**Discussion by pythcon** [wg-easy/wg-easy#301] [Ideas]
Is there a flag / env variable to change the mtu size for the “server”? When setting WG_MTU, that only adjusts the value for new clients. If I manually edit the wg0.conf file and add the MTU = 1300 value, it works fine. The problem with this fix is that it will be overwritten if the container is restarted.
Thanks!

**Comment by bcookatpcsd** [comment]
I wanted to share this link: https://gist.github.com/nitred/f16850ca48c48c79bf422e90ee5b9d95

The author does many tests to achieve his optimal MTU setting.  (1384 in his tests)

I use 1384 for my 1500 mtu interface.. 

https://keremerkan.net/posts/wireguard-mtu-fixes/

This article goes through some things.. and requires a re-read and understanding clamping-mss.. but logically the author has stepped through figuring out what works best for him.. 

The author might have a input switch for 'what is your egress MTU'.. (maybe you have a PPPoE uplink.. )

**Comment by aayushbhatia06** [comment]
Did you ever find a way to make it happen? Thanks

**Comment by bcookatpcsd** [comment]
yes I did.. 

```
cat -n wg-easy.sh
     1  docker pull weejewel/wg-easy
     2  docker stop wg-easy
     3  docker rm wg-easy
     4  docker run -d \
     5    --name=wg-easy \
     6    -e WG_HOST=ext.ip.add.ress \
     7    -e WG_MTU=1420 \
     8    -e WG_PERSISTENT_KEEPALIVE=25 \
     9    -e WG_DEFAULT_DNS=192.0.2.254 \
    10    -e WG_DEFAULT_ADDRESS=10.9.8.x \
    11    -e PASSWORD=yourpasswordhere(!@#$%) \
    12    -e TZ=America/New_York \
    13    -v /etc/docker/wireguard:/etc/wireguard \
    14    --network="host" \
    15    --cap-add=NET_ADMIN \
    16    --cap-add=SYS_MODULE \
    17    --restart unless-stopped \
    18    weejewel/wg-easy
    19
```

![image](https://github.com/wg-easy/wg-easy/assets/55087301/bc07d5a5-1717-4864-9936-122cad7fedc0)

![image](https://github.com/wg-easy/wg-easy/assets/55087301/0cd96a9f-2217-4812-80ed-3321b259679d)

```
ip link show wg0
8: wg0: <POINTOPOINT,NOARP,UP,LOWER_UP> mtu 1420 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1024
    link/none
```

Hope this helps..
